nscq/src/nscq_handler.rs

```diff
@@ -0,0 +1,285 @@
+use std::{
+    collections::HashMap,
+    ffi::{c_char, c_void},
+    ptr,
+    sync::{LazyLock, Mutex},
+};
+
+use crate::functions::nscq_uuid_to_label;
+
+use super::{
+    session::Session,
+    types::{
+        NscqArch, NscqAttestationCertificate, NscqAttestationReport, NscqCallback, NscqRc,
+        NscqUuid, UserData, NSCQ_ATTESTATION_REPORT_SIZE,
+    },
+};
+
+const TNVL_BIT_POSITION: usize = 0;
+const LOCK_BIT_POSITION: usize = 1;
+const UUID_PATH: &str = "/drv/nvswitch/{device}/uuid";
+const TNVL_STATUS_PATH: &str = "/config/pcie_mode";
+const ATTESTATION_REPORT_PATH: &str = "/config/attestation_report";
+const ATTESTATION_CERTIFICATE_CHAIN_PATH: &str = "/config/certificate";
+const ALL_SWITCHES: &str = "/{nvswitch}";
+const ARCH: &str = "/id/arch";
+
+static SHARED_UUID: LazyLock<Mutex<Vec<String>>> = LazyLock::new(|| Mutex::new(Vec::new()));
+static SHARED_TNVL_STATUS: LazyLock<Mutex<HashMap<String, u8>>> =
+    LazyLock::new(|| Mutex::new(HashMap::new()));
+static SHARED_ATTESTATION_REPORT: LazyLock<
```

Comment:
What's the purpose of these static variables ? Attestation reports need to be regenerated constantly (on Atoma node at least).